resources/design-document.md

```diff
@@ -0,0 +1,335 @@
+
+This initial iteration will provide the minimum loveable product (MLP) including creating, retrieving, updating and 
+  archiving translation cases and translation clients, as well as sorting these objects in useful ways.
+
+I will use API Gateway and Lambda to create seven endpoints (`GetTranslationCase`, `CreateTranslationCase`, 
+`UpdateTranslationCase`, `ArchiveTranslationCase`, `GetTranslationClient`, `UpdateTranslationClient`, 
+  `ArchiveTranslationClient`) that will handle the creation, update, retrieval and archiving of translation clients 
+  and translation cases.
+
+I will store the translation clients and translation cases in tables in DynamoDB.
+
+TranslationTracker will also provide a web interface for users to manage their translation cases and translation 
+  clients. A main page providing lists of the translation cases with the soonest upcoming deadlines and translation 
+  clients with most recent activity will allow creation of new translation cases and link off to pages per-translation 
+  client and per-translation case to update translation progress and other details.
+
+# 6. API
+
+## 6.1. Public Models
+
+```
+// TranslationCaseModel
+
+String translationCaseId;
+String translationClientId;
+String sourceTextTitle;
+String sourceTextAuthor;
+String translatedTitle;
+String caseNickname;
+Double rate;
+String rateUnit;
+Integer count;
+String countUnit;
+Double grossPayment;
+Double taxRate;
+String projectType;
+String dueDate;
+String startDate;
+String endDate;
+Boolean openCase;
+Boolean casePaid;
+Boolean rushJob;
+Map<String, Map<String, String>> progressLog;
+Double totalWorkingHours;
+Double wordsPerHour;
+
+// TranslationClientModel
+
+String translationClientId;
+String translationClientName;
+String translationClientType;
+Map<String, Map<String, String>> translationCaseHistory;
+Map<String, Map<String, String>> paymentHistory;
+Map<String, String> outstandingPayments;
+```
+
+### Stretch Goal
+
+```
+// EstimateModel
+
+String estimateId;
+String translationClientId;
+String translationClientName;
+String translationClientType;
+String sourceTextTitle;
+String sourceTextAuthor;
+Double estimatedRate;
+String estimatedRateUnit;
+Integer count;
+String countUnit;
+Double estimatedGrossPayment;
+Double taxRate;
+String projectType;
+String dueDate;
+Double totalWorkingHoursEstimate;
+Double wordsPerHourEstimate;
+```
+
+## 6.2. Get Translation Case Endpoint
+
+* Accepts `GET` requests to `/cases/:id`
+* Accepts a translation case ID and returns the corresponding `TranslationCaseModel`.
+    * If the given translation case ID is not found, will throw a
+      `TranslationCaseNotFoundException`
+
+<img src="images/GetTranslationCase-Sequence.png" width="100%" alt="Get Translation Case Endpoint Sequence Diagram"/>
+
+## 6.3 Create Translation Case Endpoint
+
+* Accepts `POST` requests to `/cases`
+* Accepts data to create a new `TranslationCase` with a provided case nickname, a given translation case ID, with all 
```

Comment:
Does "given translation case ID" mean it's something the user enters? How is that used vs the "unique translation case ID" that's assigned by the app?

Reply:
This highlighted a bigger design issue.
Addressed this specific issue by now only requiring the caseNickname.